src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -170,8 +175,34 @@ private static int PublishHost(ProjectContext context, string outputPath)
 
             // Copy the host
             File.Copy(hostPath, outputExe, overwrite: true);
+
+            // Publish the runtime
+            var runtimePath = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, "..", "runtime", "coreclr"));
+            if(!Directory.Exists(runtimePath))
```

Comment:
I think the right approach is if the runtime is published already via a direct reference then dont overwrite it.